commands/addBlockchain.ts

```diff
@@ -43,46 +43,55 @@ export async function execute(interaction: CommandInteraction) {
   }
 
   await interaction.reply({
-    content: `Are you sure you want to add ${blockchain.trim()}?`,
+    content: `Are you sure you want to add \`${blockchain.trim()}\`?`,
     components: [buttonRow],
     ephemeral: true,
   });
 
-  const buttonReply = await interaction.channel?.awaitMessageComponent({ componentType: 'BUTTON' });
-  if (!buttonReply) {
-    return;
-  }
+  const interactionMessage = await interaction.fetchReply();
 
-  const buttonSelected = buttonReply.customId;
-  buttonReply.update({ components: [] });
-  if (buttonSelected === REPLY.NO) {
-    buttonReply.followUp({
-      content: `"${blockchain.trim()}" was not added`,
-      ephemeral: true,
-    })
-    return;
-  }
-  else {
-    try {
-      const foundChain = await findBlockchainByName(blockchain.trim());
-      if (foundChain) {
-        await interaction.editReply('This blockchain is already registered.');
-      }
-      else {
-        await createBlockchain(blockchain.trim(), website ? website.trim() : website);
-        await interaction.editReply('Thank you. The blockchain has been added.');
-      }
+  if (interactionMessage instanceof Message) {
```

Comment:
I also wondered what's the other types of interactionMessage and why we need this.

Reply:
`const interactionMessage: Message<true> | APIMessage | Message<boolean>`